include/libopencm3/stm32/common/sdio_common_all.h

```diff
@@ -0,0 +1,266 @@
+#define SDIO_ICR			MMIO32(SDIO_BASE + 0x38) /* SDIO interrupt clear register (SDIO_ICR) */
+#define SDIO_MASK			MMIO32(SDIO_BASE + 0x3C) /* SDIO mask register (SDIO_MASK) */
+#define SDIO_FIFOCNT		MMIO32(SDIO_BASE + 0x48) /* SDIO FIFO counter register (SDIO_FIFOCNT) */
+#define SDIO_FIFO			MMIO32(SDIO_BASE + 0x80) /* SDIO data FIFO register (SDIO_FIFO) */
+
+/* --- SDIO_POWER values -------------------------------------------------- */
+#define SDIO_POWER_PWRCTRL_SHIFT		(uint32_t)0
+#define SDIO_POWER_PWRCTRL_MASK	    	(uint32_t)(0x3 << SDIO_POWER_PWRCTRL_SHIFT)
+#define SDIO_POWER_PWRCTRL_PWROFF		(uint32_t)(0x0 << SDIO_POWER_PWRCTRL_SHIFT)
+#define SDIO_POWER_PWRCTRL_PWRON		(uint32_t)(0x3 << SDIO_POWER_PWRCTRL_SHIFT)
+
+/* --- SDIO_POWER values -------------------------------------------------- */
+#define SDIO_CLKCR_HWFC_EN				(uint32_t)(1 << 14)
+#define SDIO_CLKCR_NEGEDGE				(uint32_t)(1 << 13)
+#define SDIO_CLKCR_WIDBUS_SHIFT			(uint32_t)11
+#define SDIO_CLKCR_WIDBUS_MASK      	(uint32_t)(0x3 << SDIO_CLKCR_WIDBUS_SHIFT)
+#define SDIO_CLKCR_BYPASS				(uint32_t)(1 << 10)
+#define SDIO_CLKCR_PWRSAV				(uint32_t)(1 << 9)
+#define SDIO_CLKCR_CLKEN				(uint32_t)(1 << 8)
+#define SDIO_CLKCR_CLKDIV_MSK			(uint32_t)(0xFF << 0)
+#define SDIO_CLKCR_CLEAR_MASK         	((uint32_t)0xFFFF8100)
+
+/* --- SDIO_CMD values ---------------------------------------------------- */
+#define SDIO_CMD_ATACMD					(uint32_t)(1 << 14)
+#define SDIO_CMD_NIEN					(uint32_t)(1 << 13)
+#define SDIO_CMD_ENCMDCOMPL				(uint32_t)(1 << 12)
+#define SDIO_CMD_SDIOSUSPEND			(uint32_t)(1 << 11)
+#define SDIO_CMD_CPSMEN					(uint32_t)(1 << 10)
+#define SDIO_CMD_WAITPEND				(uint32_t)(1 << 9)
+#define SDIO_CMD_WAITINT				(uint32_t)(1 << 8)
+#define SDIO_CMD_WAITRESP_SHIFT			(uint32_t)6
+#define SDIO_CMD_WAITRESP_MSK       	(uint32_t)(0x3 << SDIO_CMD_WAITRESP_SHIFT)
+#define SDIO_CMD_CMDINDEX_MSK			(uint32_t)(0x3F << 0)
+#define SDIO_CMD_CLEAR_MASK             ((uint32_t)0xFFFFF800)
+
+/* --- SDIO_RESPCMD values ------------------------------------------------ */
+#define SDIO_RESPCMD_MSK				(uint32_t)(0x3F << 0)
+
+/* --- SDIO_DLEN values --------------------------------------------------- */
+#define SDIO_DLEN_MSK			    	(uint32_t)0x1FFFFFF
+
+/* --- SDIO_DCTRL values -------------------------------------------------- */
+#define SDIO_DCTRL_SDIOEN				((uint32_t)(1 << 11))
+#define SDIO_DCTRL_RWMOD				((uint32_t)(1 << 10))
+#define SDIO_DCTRL_RWSTOP				((uint32_t)(1 << 9))
+#define SDIO_DCTRL_RWSTART				((uint32_t)(1 << 8))
+#define SDIO_DCTRL_DBLOCKSIZE_SHIFT		((uint32_t)4)
+#define SDIO_DCTRL_DBLOCKSIZE_MSK		((uint32_t)(0xF << SDIO_DCTRL_DBLOCKSIZE_SHIFT))
+#define SDIO_DCTRL_DMAEN				((uint32_t)(1 << 3))
+#define SDIO_DCTRL_DTMODE				((uint32_t)(1 << 2))
+#define SDIO_DCTRL_DTDIR				((uint32_t)(1 << 1))
+#define SDIO_DCTRL_DTEN					((uint32_t)(1 << 0))
+#define SDIO_DCTRL_CLEAR_MASK           ((uint32_t)0xFFFFFF08)
+
+typedef enum
```

Comment:
There's far much namespace pollution via all the enums you've typedeffed.  Are they really all necessary?  Can some of them be done via namespaced defines or sensible arguments?  If not, drop teh typedef, and name the enums with better prefixes to keep them within sdio